slife:
  rate: 1
  synth_pcl_size: 10
  topics:
    pointcloud: /camera/depth/pointcloud
    debug_grid: /slife/debug/grid
    debug_1: /slife/debug/history
    debug_2: /slife/debug/debug_2
    estimate: /slife/estimate
  landscape:
    measure_radius: 0.7
    smooth_radius: 0.001
    precision: 1
  target_optimization_group: pose # options: {position, quaternion_r4, pose, dual_quaternion [not implemented yet]}, warning: need to recompile with `using TargetGroup = <desired target group>`
  optimizer:
    step_sizes: [0.02, 0.015]
    norm_weights: [1, 1]
    threshold: 0.000
    max_iterations: 100
    initialization_type: identity # options: {identity}
    record_history: true  # time consuming, use only for debug
    cost:
      batch_size: 10
  debug:
    test_grid:
      min: -1.0
      max: 1.0
      step: 0.06
    z_test_value: 0.
    test_type: none # options: {none, landscape_value, landscape_gradient, cost_value, cost_gradient}